pypesto/collections/collections.py

```diff
@@ -0,0 +1,497 @@
+import numpy as np
+import pandas as pd
+from typing import Sequence, Tuple, Callable, Dict
+
+from ..prediction import PredictionResult, PredictionConditionResult
+from .constants import (PREDICTOR, PREDICTION_ID, PREDICTION_RESULTS,
+                        PREDICTION_ARRAYS, PREDICTION_SUMMARY, OUTPUT,
+                        OUTPUT_SENSI, TIMEPOINTS, X_VECTOR, NX, X_NAMES,
+                        NVECTORS, VECTOR_TAGS, PREDICTIONS, MODE_FUN,
+                        CollectionType, COLLECTION_TYPE, MEAN, MEDIAN,
+                        STANDARD_DEVIATION, PERCENTILE, SUMMARY, LOWER_BOUND,
+                        UPPER_BOUND)
+
+
+class CollectionPrediction:
+    """
+    A collection prediction consists of a collection, i.e., a set of parameter
```

Comment:
`set`: In the python context I'd assume this collection to be unordered and having unique parameter vectors, which is not necessarily the case, right?

Reply:
In principle, there is no ordering assumed. Unique parameters vectors: not necessarily.... There may be applications where non-uniqueness may be helpful, so I don't want to enforce it...